### PR TITLE
Improve inline properties feature

### PR DESCRIPTION
**Pull request by Egorand**
Based on the discussion on #245

### CODE REVIEW

File: src/main/java/com/squareup/kotlinpoet/PropertySpec.kt [hunk 1]
```diff
@@ -149,12 +156,22 @@ class PropertySpec private constructor(builder: Builder) {
...
     }
 
     fun setter(setter: FunSpec) = apply {
       require(setter.name == SETTER) { "${setter.name} is not a setter" }
       check(this.setter == null) { "setter was already set" }
       this.setter = setter
+      checkPromoteInlineToProperty()
+    }
+
+    private fun checkPromoteInlineToProperty() {
```
Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/PropertySpec.kt]: Can you do this at render time instead?

Comment (swankjesse) [reply src/main/java/com/squareup/kotlinpoet/PropertySpec.kt]: Ie. keep `modifiers` as-is as a property of this class, but add `inline` when printing it? That way if anyone uses KotlinPoet as a basic AST they get `inline` where they expect it

Comment (JakeWharton) [reply src/main/java/com/squareup/kotlinpoet/PropertySpec.kt]: And (more importantly in my opinion) they don't see `inline` where they didn't specify it.

Comment (Egorand) [reply src/main/java/com/squareup/kotlinpoet/PropertySpec.kt]: Agree, good point. Force-pushed an updated version.

File: src/main/java/com/squareup/kotlinpoet/PropertySpec.kt [hunk 2]
```diff
@@ -52,14 +55,19 @@ class PropertySpec private constructor(builder: Builder) {
       codeWriter.emitCode("%[%L%]", initializer)
     }
     if (!inline) codeWriter.emit("\n")
+    val implicitAccessorModifiers = if (isInlineProperty) {
```
Comment (swankjesse) [inline src/main/java/com/squareup/kotlinpoet/PropertySpec.kt]: Yeah, I really like this

**Comment by swankjesse** [comment]
I really like where this landed. Thank you.

**Pull request merged**